src/Grpc.AspNetCore.Server/Internal/UnaryServerCallHandler.cs

```diff
@@ -57,13 +57,18 @@ public override async Task HandleCallAsync(HttpContext httpContext)
             // Activate the implementation type via DI.
             var activator = httpContext.RequestServices.GetRequiredService<IGrpcServiceActivator<TService>>();
             var service = activator.Create();
+            var streamWriter = new HttpContextStreamWriter<TResponse>(httpContext, Method.ResponseMarshaller.Serializer);
```

Comment:
Why did you add this?

Reply:
Because of WriteOptions. The service can modify the WriteFlags during the call and it can affect how TResponse should be written. For example, the compress flag can be set. Also this standardizes this portion of the logic for all handlers.

Reply:
Ok. Then we should have a write helper method that takes the serializer, write options and message. The StreamWriter and this place can use it. That avoids us allocating the stream writer here.